hypar-io/Elements
# Don't allocate lines during testing in constructor.

BACKGROUND:
- While reviewing #735, I noticed that our validity tests in Polygon constructors allocate a lot of `Line` instances. This is not required as most tests simply need a start and end vector describing a segment.

DESCRIPTION:
- This PR replaces areas where we construct `Line` instances for testing with uses of `Edges()`, and it adds `Polyline.Edges` so we can use edges there as well.

TESTING:
- See the benchmarks below for speed/memory improvements.
- See polygon and polyline tests which continue to pass.
  
FUTURE WORK:
- Is there any future work needed or anticipated?  Does this PR create any obvious technical debt?

REQUIRED:
- [x] All changes are up to date in `CHANGELOG.md`.

Before
|                Method | SkipValidation |     Mean |    Error |   StdDev |     Gen 0 |     Gen 1 | Allocated |
|---------------------- |--------------- |---------:|---------:|---------:|----------:|----------:|----------:|
| 'Draw all HSS beams.' |          False | 31.71 ms | 0.628 ms | 1.032 ms | 2000.0000 | 1000.0000 |     12 MB |
| 'Draw all HSS beams.' |           True | 31.60 ms | 0.626 ms | 1.144 ms | 2000.0000 | 1000.0000 |     10 MB |

After
|                Method | SkipValidation |     Mean |    Error |   StdDev |     Gen 0 |     Gen 1 | Allocated |
|---------------------- |--------------- |---------:|---------:|---------:|----------:|----------:|----------:|
| 'Draw all HSS beams.' |          False | 32.13 ms | 0.628 ms | 1.116 ms | 2000.0000 | 1000.0000 |     11 MB |
| 'Draw all HSS beams.' |           True | 30.52 ms | 0.596 ms | 0.816 ms | 2000.0000 | 1000.0000 |     10 MB |


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/hypar-io/elements/737)
